website/content/en/preview/tasks/amitasks.md

```diff
@@ -0,0 +1,178 @@
+For example, you could have:
+
+* **Test clusters**: On these private clusters, you can run the latest AMIs for your workloads in a safe environment.
+* **Production clusters**: When you feel that everything is working properly, you can set the latest AMIs to be deployed in your production clusters so they are note upgraded.
+
+Remember that it is still best practice to gradually roll new AMIs into your cluster, even if they have been tested.
+
+### Task 2: Lock down which AMIs are selected
+
+Instead of letting Karpenter always run the latest AMI, you can change Karpenter’s default behavior.
+When you configure the [**EC2NodeClass**]({{< relref "../concepts/nodeclasses" >}}), you can set a specific AMI that you want Karpenter to always choose, using the `amiSelectorTerms` field.
+This prevents a new and potentially untested AMI from replacing existing nodes when those nodes are terminated.
+
+With the `amiSelectorTerms` field in an `EC2NodeClass`, you can set a specific AMI for Karpenter to use, based on AMI name or id (only one is required).
+These examples show two different ways to identify the same AMI:
+
+```bash
+amiSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: "${CLUSTER_NAME}"
+      environment: prod
+  - name: al2023-ami-2023.3.20240219.0-kernel-6.1-x86_64
+```
+
+```bash
+amiSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: "${CLUSTER_NAME}"
+      environment: prod
+  - id: ami-052c9ea013e6e3567
+```
+
+See the [**spec.amiSelectorTerms**]({{< relref "../concepts/nodeclasses/#specamiselectorterms" >}}) section of the NodeClasses page for details. 
+Keep in mind, that this could prevent you from getting critical security patches when new AMIs are available, but it does give you control over exactly which AMI is running.
+
+
+### Task 3: Control the pace of node disruptions
+
+To help prevent the possibilities of a new AMI being deployed to all your nodes and breaking all of your workloads, you can enable Karpenter [**Disruption Budgets**]({{< relref "../concepts/disruption/#disruption-budgets " >}}).
+Disruption Budgets limit when and to what extent nodes can be disrupted.
+You can prevent disruption based on nodes (a percentage or number of nodes that can be disrupted at a time) and schedule (excluding certain times from disrupting nodes).
+You can set Disruption Budgets in a `NodePool` spec.
+Here is an example:
+
+```bash
+disruption:
+  consolidationPolicy: WhenEmpty
+  consolidateAfter: 30s
+  expireAfter: 1440h
+  budgets:
+  - nodes: 15%
+  - nodes: "3"
+  - nodes: "0"
+    schedule: "0 7 * * sat-sun"
+    duration: 12h
+```
+
+The `disruption` settings define a few fields that indicate the state of a node that should be disrupted.
+The `consolidationPolicy` field indicates that a node should be disrupted if the node is either empty (`WhenEmpty`) or not running any pods (`WhenEmpty`).
```

Comment:
I'm wondering if, for the purposes of this example, we should be using `WhenUnderutilized`. I think that's what we expect most users to be using and is also where you are at the highest risk of existing workloads being upgraded, as they are migrated from the node which was consolidated to the new node. This would drop `consolidateAfter` entirely for the example but I think that's fine.

Reply:
Done.